diff -aurN a/lib/libjpeg-80/transupp.c b/lib/libjpeg-80/transupp.c
--- a/lib/libjpeg-80/transupp.c	2013-03-01 15:00:12.951609495 +0800
+++ b/lib/libjpeg-80/transupp.c	2013-04-14 17:04:53.323653449 +0800
@@ -850,6 +850,23 @@
     info->output_height = MCU_rows * info->iMCU_sample_height;
 }
 
+/*
+ * Compute output image dimensions and related values.
+ * NOTE: this is exported for possible use by application.
+ * Hence it mustn't do anything that can't be done twice.
+ */
+
+EXTERN(void)
+jpeg_core_output_dimensions (j_decompress_ptr cinfo)
+/* Do computations that are needed before master selection phase.
+ * This function is used for transcoding and full decompression.
+ */
+{
+  /* Hardwire it to "no scaling" */
+  cinfo->output_width = cinfo->image_width;
+  cinfo->output_height = cinfo->image_height;
+}
+
 
 /* Request any required workspace.
  *
